### PR TITLE
Nominal diff time

### LINKED ISSUE #80: toUrlPiece for NominalDiffTime loses precision
Additionally, `fromUrlPiece` for `NominalDiffTime` can only accept integers, so it can't represent all `NominalDiffTime`s.

Since `NominalDiffTime` uses `Pico` from `Data.Fixed` [underneath](http://hackage.haskell.org/package/time-1.9.2/docs/src/Data.Time.Clock.Internal.NominalDiffTime.html#NominalDiffTime), perhaps this can be changed once https://github.com/fizruk/http-api-data/issues/77 is implemented?

### PR DESCRIPTION
**Pull request by rockbmb**
Closes #80.

### CODE REVIEW

**Comment by rockbmb** [comment]
@fizruk

File: src/Web/Internal/HttpApiData.hs [hunk 1]
```diff
@@ -640,7 +647,7 @@ instance FromHttpApiData ZonedTime where parseUrlPiece = runAtto Atto.zonedTime
 -- Right 2015-10-03 00:14:24 UTC
 instance FromHttpApiData UTCTime   where parseUrlPiece = runAtto Atto.utcTime
 
-instance FromHttpApiData NominalDiffTime where parseUrlPiece = fmap fromInteger . parseUrlPiece
+instance FromHttpApiData NominalDiffTime where parseUrlPiece = fmap (realToFrac :: F.Pico -> NominalDiffTime) . parseUrlPiece
```
Comment (fizruk) [inline src/Web/Internal/HttpApiData.hs]: I believe `realToFrac` is not as efficient as `secondsToNominalDiffTime`.
Can you do the same trick as with `nominalDiffTimeToSeconds`?

Comment (fizruk) [inline src/Web/Internal/HttpApiData.hs]: I believe `realToFrac` is not as efficient as `secondsToNominalDiffTime`.
Can you do the same trick as with `nominalDiffTimeToSeconds`?

**Comment by rockbmb** [comment]
@fizruk thanks for pointing that out, should be better now.

**Comment by phadej** [comment]
@rockbmb could you also change the test generator to e.g.
```haskell
oneof
    [ fromInteger <$> arbitrary -- integral
    , secondsToNominalDiffTime <$> arbitrary -- with decimal part
    ]
```

The CPP have (unfortunately) to be repeated in tests too. :/ (we shouldn't export that function from the public API)

**Comment by rockbmb** [comment]
@phadej I'm not sure understand, where would this code be used?

**Comment by rockbmb** [comment]
@phadej Could you clarify my comment ⏫ please?

**Comment by phadej** [comment]
@rockbmb you should change https://github.com/fizruk/http-api-data/blob/283c7474cbbeb3e586387312dc077b51641a5350/test/Web/Internal/HttpApiDataSpec.hs#L142

so it would fail without your change to the instance

**Comment by phadej** [comment]
@fizruk, note, that this change to instances will be (subtle) breaking change. I consider the `ToHttpApiData` breaking. The more lenient `FromHttpApiData` alone would been probably just minor change, though some might rely on the parsers accepting only integers.

**Comment by rockbmb** [comment]
@phadej should be fixed now.

**Comment by phadej** [comment]
Merged as part of #89 . Thanks!

**Pull request closed without merging**